### PR TITLE
[Documentation]: Polish Ecephys and NWBFile Metadata Best Practices

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Feel free to make any changes

### CODE REVIEW

File: docs/best_practices/ecephys.rst [hunk 1]
```diff
@@ -2,42 +2,74 @@ Extracallular electrophysiology
...
+.. _best_practice_negative_spike_times:
+
+Negative Spike Times
+~~~~~~~~~~~~~~~~~~~~
+
+All spike times should be greater than zero. Being less than zero implies the spikes are either trial-aligned (and
+should therefore be aligned to the global :py:attr:`~pynwb.file.NWBFile.timestamps_reference_time`) or the
+:py:attr:`~pynwb.file.NWBFile.timestamps_reference_time` is not set to agree with the
+:py:attr:`~pynwb.file.NWBFile.session_start_time`. In either case the spike times (and all other temporal data) should
+be aligned to the earliest recorded timestamp in the file.
```
Comment (bendichter) [inline docs/best_practices/ecephys.rst]:
```suggestion
should therefore be after the :py:attr:`~pynwb.file.NWBFile.timestamps_reference_time`.
```

`timestamps_reference_time` does not have to agree with `session_start_time`.

**Comment by CodyCBakerPhD** [comment]
> timestamps_reference_time does not have to agree with session_start_time.

Right, that is something I keep seeing pop up - what are some common situations for decoupling the two?

**Comment by CodyCBakerPhD** [comment]
To-Do:

- [x] swap all :py:meth:`~pynwb...` to NWB Schema refs

File: docs/best_practices/nwbfile_metadata.rst [hunk 2]
```diff
@@ -1,83 +1,192 @@
...
+with a recording session. Sometimes you may find yourself having multiple NWBFiles that correspond to the same session.
+This can happen, for instance, if you separate out processing steps across multiple files or if you want to compare
+different processing outputs. In this case, the ``session_id`` should be the same for each file. Each lab should follow
+a standard structure for their own naming schemes so that sessions are unique within the lab and the IDs are easily
+human-readable.
+
+The ``identifier`` tag should be a globally unique value for the :nwb-schema:ref:`sec-NWBFile`. Two different NWBFiles
+from the same session should have different ``identifier`` values if they differ in any way. It is recommended that you
+use a well-established algorithmic generator such as :uuid:`uuid <>` (for PyNWB) or ?? (for MatNWB) to ensure
+uniqueness. The ``identifier`` does not need to be easily human-readable.
```
Comment (CodyCBakerPhD) [inline docs/best_practices/nwbfile_metadata.rst]: I noticed MATLAB does have a `uuid` package of some sort, but I couldn't find example usage via MatNWB documentation; what should our recommendation be?

**Review by bendichter — APPROVED** [review]
looks great!

**Pull request merged**